### PR TITLE
feat: Add ViewComponent & Convert Notification to ViewComponent

### PR DESCRIPTION
**Pull request by crespire**
### What changed, and why?
* Adds ViewComponent gem
* Updates test helper to include ViewComponent helpers
* Updates test factories to facilitate future notification testing
* Updates `BaseNotification` to better serve as an abstract class
* Updates `FollowupNotification` with render helper methods
* Deletes `_notification.html.erb` partial
* Adds a `NotificationComponent` view component
* Consolidates display logic into ViewComponent
* Updates `NotificationsHelper` to remove UI related helpers, which have been moved into the view component, supported by new methods on `BaseNotification` and `FollowupNotification`

### How will this affect user permissions?
- Volunteer permissions: No changes
- Supervisor permissions: No changes
- Admin permissions: No changes

### How is this tested? (please write tests!) 💖💪
New `NotificationComponent` spec added and replicates previous tests. Replicated tests from previous specs removed.

### Screenshots please :)
Before: 
image

After:
image

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

File: app/components/notification_component.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class NotificationComponent < ViewComponent::Base
```
Comment (dpaola2) [inline app/components/notification_component.rb:3]: `attr_reader :notification`

File: app/components/notification_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<a
```
Comment (dpaola2) [inline app/components/notification_component.html.erb:1]: replace `@notification` with `notification`

File: app/components/notification_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  href="<%= @notification.url %>"
+  class="<%= @notification.read? ? "bg-light text-muted " : "" %>list-group-item list-group-item-action flex-column align-items-start">
```
Comment (dpaola2) [inline app/components/notification_component.html.erb:3]: ```class="<%= muted_class %>"```

```
def muted_class
  if notification.read?
    'bg-light text-muted'
  else
    ''
  end
end
```

File: app/components/notification_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <div class="d-flex flex-row">
+    <div class="d-flex-shrink-0"><%= @notification.read? ? "" : sanitize("<i class='fas fa-bell'></i>", tags: %w(i), attributes: %w(class)) %></div>
```
Comment (dpaola2) [inline app/components/notification_component.html.erb:5]:
```
<% if !@notification.read? %>
  <put it here>
<% end %>
```

**Comment by crespire** [comment]
@FireLemons can this get merged in?

**Comment by dpaola2** [comment]
@crespire can you push an empty commit to see if that will fix the docker spec?

**Comment by crespire** [comment]
Updated to merge in new theme and changes from `main`:

image
image

**Comment by crespire** [comment]
@dpaola2 Ready for review

**Review by dpaola2 — APPROVED** [review]

**Pull request merged**